### PR TITLE
dotnet: simplify tests

### PR DESCRIPTION
**Pull request by matklad**
One issue I've fond when running tests on widows is that the data file isn't actually deleted, with an error message saying that another process still uses the data file.

Let's try:

- to always wait for the process
- and also don't swallow the error

### CODE REVIEW

File: src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -1626,16 +1626,9 @@ public TBServer()
 
     public void Dispose()
     {
-        try
-        {
-            if (process != null && !process.HasExited)
```
Comment (matklad) [inline src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:1631]: @batiati process can't be null here, right? If we are running Dispose, constructor has finished?

Or is there some special C# semantics when Dispose might get called if you only managed to run through half of the constructor's code?

Comment (batiati) [reply src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs:1631]: Nice catch! Since `process` is readonly, it cannot be null after the constructor
https://github.com/tigerbeetle/tigerbeetle/blob/2cdabe44b44a54d279e2866da6812e1201506eb5/src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs#L1624

And objects that are not fully constructed don't run the finalizer/destructor/dispose.

I suppose the original intent was to prevent double `Dispose(),` but it's completely out of scope to cover this edge case in a test runner!

**Review by batiati — COMMENTED** [review]
Just a nit, change the type from `String` to `string` here:

https://github.com/tigerbeetle/tigerbeetle/blob/2cdabe44b44a54d279e2866da6812e1201506eb5/src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs#L1625-L1626

**Review by batiati — APPROVED** [review]
